tools/cli/util.go

```diff
@@ -743,26 +808,32 @@ func parseTimeDuration(duration string) (dur time.Duration, err error) {
 	return
 }
 
-func strToTaskListType(str string) s.TaskListType {
+func strToTaskListType(str string) *types.TaskListType {
 	if strings.ToLower(str) == "activity" {
-		return s.TaskListTypeActivity
+		return types.TaskListTypeActivity.Ptr()
 	}
-	return s.TaskListTypeDecision
+	return types.TaskListTypeDecision.Ptr()
 }
 
-func getPtrOrNilIfEmpty(value string) *string {
-	if value == "" {
-		return nil
-	}
-	return common.StringPtr(value)
+func getCliIdentity() string {
+	return fmt.Sprintf("cadence-cli@%s", getHostName())
 }
 
-func getCliIdentity() string {
+// GetWorkerIdentity gets a default identity for the worker.
+//
+// This contains a random UUID, generated each time it is called, to prevent identity collisions when workers share
+// other host/pid/etc information.  These alone are not guaranteed to be unique, especially when Docker is involved.
+// Take care to retrieve this only once per worker.
+func getWorkerIdentity(tasklistName string) string {
```

Comment:
Why do we need `getWorkerIdentity`? IMO `getCliIdentity` is enough for this application and could be used everywhere.